plugins/woocommerce/src/Blocks/AssetsController.php

```diff
@@ -196,13 +196,59 @@ private function get_prerender_resource_hints() {
 		return $urls;
 	}
 
+	/**
+	 * Get the block asset resource hints in the cache or null if not found.
+	 *
+	 * @return array|null Array of resource hints.
+	 */
+	private function get_block_asset_resource_hints_cache() {
+		$cache = get_site_transient( 'woocommerce_block_asset_resource_hints' );
+
+		$current_version = array(
+			'woocommerce' => WOOCOMMERCE_VERSION,
+			'wordpress'   => get_bloginfo( 'version' ),
+		);
+
+		if ( isset( $cache['version'] ) && $cache['version'] === $current_version ) {
```

Comment:
Could this cause confusion in development builds if dependencies change during dev?